src/lib/shared/ui/format-duration.ts

```diff
@@ -14,5 +15,15 @@ export function formatDuration(durationInSeconds: number) {
   if (durationInSeconds >= 24 * 60 * 60) {
     throw Error('Durations longer or equal to 24 hours are not supported');
   }
-  return new Date(durationInSeconds * 1000).toISOString().substring(11, 19);
+
+  let sliceStart = 11;
+  if (durationInSeconds < 10 * 60 * 60) {
+    sliceStart++;
+
+    if (durationInSeconds < 1 * 60 * 60) {
+      sliceStart += 2;
+    }
+  }
+
+  return new Date(durationInSeconds * 1000).toISOString().substring(sliceStart, 19);
```

Comment:
To be honest, this feels like a really dirty workaround to avoid using `getHours`/`getMinutes`/`getSeconds`. At least you could've used RegEx to extract the time from the ISO string.
The solution I would prefer is to use `new Intl.DateTimeFormat('en-GB', { timeStyle: 'medium' }).format(date)` (from [MDN - `Intl.DateTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat))

Reply:
It's not ideal since it leaves the leading zero in the hours, that's not exactly something you'd want to see in a duration representation. I agree that this is hacky, but doing something with a bunch of `if` statements felt like overengineering to me. Idk how to improve this

Reply:
A solution that does the above (with `Intl`) and then checks if `duration.startsWith('00:')` and then either `duration.slice(3)` or even just `duration.replace(/^00:/, '')` without the check would be a lot less hacky than the `sliceStart` thing, and even be much shorter.

Reply:
But you also need to cut out the leading zero if the hour count is 1-9: `1:00:00`, not `01:00:00`

Reply:
And honestly, now that I think about it, we should have done it with minutes too. Okay, this entire function should be rewritten properly. I'll make an issue to do that (#72)

Reply:
Then change the RegEx to `^[0:]+` and it should handle that case as well